arm/Microsoft.Network/privateEndpoints/deploy.bicep

```diff
@@ -0,0 +1,109 @@
+  'Managed Application Contributor Role': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','641177b8-a67a-45b9-a033-47bc880bb21e')
+  'Managed Application Operator Role': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','c7393b34-138c-406f-901b-d8cf2b17e6ae')
+  'Managed Applications Reader': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','b9331d33-8a36-4f8c-b097-4f54124fdb44')
+  'Monitoring Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','749f88d5-cbae-40b8-bcfc-e573ddc772fa')
+  'Monitoring Metrics Publisher': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','3913510d-42f4-4e42-8a64-420c390055eb')
+  'Monitoring Reader': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','43d0d8ad-25c7-4714-9337-8ba259a9fe05')
+  'Network Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','4d97b98b-1d4f-4787-a291-c67834d212e7')
+  'Resource Policy Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','36243c78-bf99-498c-9df9-86d9f8d28608')
+  'User Access Administrator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions','18d7d88d-d35e-4fb5-a5c3-7773c20a72d9')
+}
+
+module pid_cuaId './.bicep/nested_pid.bicep' = if (!empty(cuaId)) {
+  name: 'pid-${cuaId}'
+  params: {}
+}
+
+resource privateEndpoint 'Microsoft.Network/privateEndpoints@2021-02-01' = {
+  name: privateEndpointName
+  location: location
+  tags: tags
+  properties: {
+    privateLinkServiceConnections: [
+      {
+        name: privateEndpointName
+        properties: {
+          privateLinkServiceId: serviceResourceId
+          groupIds: groupId
+        }
+      }
+    ]
+    manualPrivateLinkServiceConnections: []
+    subnet: {
+      id: targetSubnetId
+    }
+    customDnsConfigs: []
+  }
+
+  resource privateDnsZoneGroup 'privateDnsZoneGroups@2021-02-01' = if (!empty(privateDNSId)) {
+    name: 'default'
+    properties: {
+      privateDnsZoneConfigs: [
+        {
+          name: privateEndpointName
+          properties: {
+            privateDnsZoneId: privateDNSId
+          }
+        }
+      ]
+    }
+  }
+}
+
+resource privateEndpoint_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lockForDeletion) {
+  name: '${privateEndpoint.name}-doNotDelete'
+  properties: {
+    level: 'CanNotDelete'
+  }
+  scope: privateEndpoint
+}
+
+module privateEndpoint_rbac './.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: 'rbac-${deployment().name}${index}'
+  params: {
+    roleAssignmentObj: roleAssignment
+    builtInRoleNames: builtInRoleNames
+    resourceName: privateEndpoint.name
+  }
+}]
+
+output privateEndpointResourceGroup string = resourceGroup().name
+output privateEndpointResourceId string = privateEndpoint.id
+output privateEndpointName string = privateEndpointName
```

Comment:
Same as the other 2: should we go with `privateEndpointName` or `privateEndpoint.name`?